desc/magnetic_fields.py

```diff
@@ -1419,12 +1436,46 @@ def field_line_integrate(
     def odefun(rpz, s):
         rpz = rpz.reshape((3, -1)).T
         r = rpz[:, 0]
+        z = rpz[:, 2]
+        # if bounds are given, will decay the magnetic field line eqn
+        # RHS if the trajectory is outside of bounds to avoid
+        # integrating the field line to infinity, which is costly
+        # and not useful in most cases
+        decay_factor = jnp.where(
+            jnp.array(
+                [
+                    jnp.less(r, bounds_R[0]),
+                    jnp.greater(r, bounds_R[1]),
+                    jnp.less(z, bounds_Z[0]),
+                    jnp.greater(z, bounds_Z[1]),
+                ]
+            ),
+            1e-2
+            * jnp.array(
+                [
+                    # we mult by 1e-2 to accelerate the decay so that the integration
+                    # is stopped soon after the bounds are exited.
+                    jnp.exp(-((r - bounds_R[0]) ** 2)),
+                    jnp.exp(-((r - bounds_R[1]) ** 2)),
+                    jnp.exp(-((z - bounds_Z[0]) ** 2)),
+                    jnp.exp(-((z - bounds_Z[1]) ** 2)),
+                ]
+            ),
+            1.0,
+        )
+        # multiply all together, the conditions that are not violated
+        # are just one while the violated ones are continous decaying exponentials
+        decay_factor = jnp.prod(decay_factor)
```

Comment:
does this need to be along an axis? Will this affect all field lines or just the one that leaves?